examples/walkthrough/main.go

```diff
@@ -129,6 +140,114 @@ func CreatePublicSharedLink(ctx context.Context, accessGrant, bucketName, object
 	return url, nil
 }
 
+func MultipartUpload(ctx context.Context, accessGrant, bucketName, objectKey string, contents1, contents2 []byte) error {
+	// Create the project and ensure the bucket exists
+	project, err := CreateProjectAndConfirmBucket(ctx, accessGrant, bucketName)
+	if err != nil {
+		return fmt.Errorf("could not create project: %v", err)
+	}
+	defer project.Close()
+
+	// Create multipart upload
+	uploadInfo, err := project.BeginUpload(ctx, bucketName, objectKey, &uplink.UploadOptions{
+		// It's possible to set an expiration date for data.
+		Expires: time.Now().Add(defaultExpiration),
+	})
+
+	if err != nil {
+		return fmt.Errorf("unable to begin multipart upload: %v", err)
+	}
+
+	// Start the upload of part 1
+	part1, err := project.UploadPart(ctx, bucketName, objectKey, uploadInfo.UploadID, 1)
+	if err != nil {
+		return fmt.Errorf("unable to upload part 1: %v", err)
+	}
+
+	// Write the contents to part 1 upload
+	_, err = part1.Write(contents1)
+	if err != nil {
+		return fmt.Errorf("unable to write to part 1: %v", err)
+	}
+
+	// Commit part 1 upload
+	err = part1.Commit()
+	if err != nil {
+		return fmt.Errorf("unable to commit to part 1: %v", err)
+	}
+
+	// Start the upload of part 2
+	part2, err := project.UploadPart(ctx, bucketName, objectKey, uploadInfo.UploadID, 2)
+	if err != nil {
+		return fmt.Errorf("unable to upload part 2: %v", err)
+	}
+
+	// Write the contents to part 2 upload
+	_, err = part2.Write(contents2)
+	if err != nil {
+		return fmt.Errorf("unable to write to part 2: %v", err)
+	}
+
+	// Commit part 2 upload
+	err = part2.Commit()
+	if err != nil {
+		return fmt.Errorf("unable to commit to part 2: %v", err)
+	}
+
+	// Complete the multipart upload
+	multipartObject, err := project.CommitUpload(ctx, bucketName, objectKey, uploadInfo.UploadID, nil)
+	if err != nil {
+		return fmt.Errorf("unable to complete the upload: %v", err)
+	}
+
+	// Initiate a download of the same object again
+	download, err := project.DownloadObject(ctx, bucketName, objectKey, nil)
+	if err != nil {
+		return fmt.Errorf("could not open multipart object: %v", err)
+	}
+	defer download.Close()
+
+	// Read everything from the download stream
+	receivedContents, err := io.ReadAll(download)
+	if err != nil {
+		return fmt.Errorf("could not read multipart data: %v", err)
+	}
+
+	// Check that the downloaded data is the same as the uploaded data.
+	uploadedData := append(contents1, contents2...)
+	if !bytes.Equal(receivedContents, uploadedData) {
+		return fmt.Errorf("got different multipart object back: %q != %q", uploadedData, receivedContents)
+	}
+
+	fmt.Fprintln(os.Stderr, "success!")
+	fmt.Fprintln(os.Stderr, "completed multipart upload:", multipartObject.Key)
+
+	return nil
+}
+
+func createMultipartData() (contents1, contents2 []byte) {
```

Comment:
Instead of this function you can use:

```
firstPart := bytes.Repeat([]byte("abcdefghijklm"), 1e7)
secondPart := bytes.Repeat([]byte("0123456789"), 1e7)
```